aptos-move/framework/supra-framework/sources/pbo_delegation_pool.move

```diff
@@ -792,17 +873,16 @@ module supra_framework::pbo_delegation_pool {
         assert!(unlock_duration > 0, error::invalid_argument(EPERIOD_DURATION_IS_ZERO));
         //Fraction denominator can not be zero
         assert!(unlock_denominator != 0, error::invalid_argument(EDENOMINATOR_IS_ZERO));
+        let numerator_length = vector::length(&unlock_numerators);
         //Fraction numerators can not be empty
         assert!(
-            vector::length(&unlock_numerators) > 0,
+            numerator_length > 0,
             error::invalid_argument(EEMPTY_UNLOCK_SCHEDULE)
         );
-        //Fraction numerators can not be zero
-        assert!(
-            !vector::any(&unlock_numerators, |e| { *e == 0 }),
-            error::invalid_argument(ESCHEDULE_WITH_ZERO_FRACTION)
-        );
-
+        //First and last numerator can not be zero
+        assert!(*vector::borrow(&unlock_numerators,0)!=0,error::invalid_argument(ESCHEDULE_WITH_ZERO_FRACTION));
```

Comment:
This function reuses the logic in `update_unlocking_schedule`. Can you please replace it with a call to `update_unlocking_schedule` or abstract the common code into a new function?

Reply:
Taken out common validation code into a separate function.